server/src/omtLinkProvider.ts

```diff
@@ -0,0 +1,173 @@
+import { dirname, isAbsolute, resolve } from 'path';
+import { DocumentLink, Position, Range } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { readFileSync } from 'fs';
+import { glob } from 'glob';
+import { WorkspaceLookup } from './workspaceLookup';
+
+const omtUriMatch = /( +["']?)(.*\.omt)/;
+const declaredImportMatch = /( +)(module:)(.*):/;
+const importMatch = /^import:/g;
+const otherDeclareMatch = /^(\w+):/g;
+
+/**
+ * Provides DocumentLinks for the imports of an OMT file.
+ */
+export default class OMTLinkProvider {
+
+    constructor(private workspaceLookup: WorkspaceLookup) { }
+
+    provideDocumentLinks(document: TextDocument): DocumentLink[] {
+        // regular path links, with or without shorthands
+        const shorthands = this.contextPaths(document);
+        return findOMTUrl(document, (uri) => replaceStart(uri, shorthands), this.workspaceLookup);
+    }
+
+    /**
+     * Resolve the target of a DocumentLink using its data. Returns undefined when unable to resolve
+     * @param data the data from a document link
+     */
+    public resolve(data?: any) {
```

Comment:
the functionname is a little bit generic. Can it be more specific?

Reply:
renamed it to 'resolveLink'
It could resolve different type of links later and I think figuring out which kind of link it is should be done in this function.